post_processing/rayleigh_diagnostics.py

```diff
@@ -3266,3 +3266,18 @@ def streamfunction(vr,vt,r,cost,order=0):
             
     return psi
 
+###### Function for reading in the checkpoint files (i.e. P,PAB,T,TAB,W,WAB,Z,ZAB)
+def checkpoint_read(chk_file_string,nr,ntheta):
+    nell=(2*ntheta)//3
+    shape = (nell,nell)
+    i,j = np.indices(shape)
+    m = np.ceil(i <= j).astype(bool)
```

Comment:
This can be simpler.

```suggestion
    m = i <= j
```